### PR TITLE
Handle Async Callable

### PR DESCRIPTION
**Pull request by hinthornw**
```
class MyAsyncCallable:
    async def __call__(self, state):
        ...
```

### CODE REVIEW

File: langgraph/utils.py [hunk 1]
```diff
@@ -148,6 +148,31 @@ def extend(
         )
 
 
+def _isasyncgencheck(thing: RunnableLike) -> bool:
+    return (
+        inspect.isasyncgenfunction(thing)
+        or inspect.isasyncgen(thing)
+        or hasattr(thing, "__call__")
```
Comment (nfcampos) [inline langgraph/utils.py:155]: This seems recursive, which I don't think is on purpose?

File: langgraph/utils.py [hunk 2]
```diff
@@ -148,6 +148,31 @@ def extend(
         )
 
 
+def _isasyncgencheck(thing: RunnableLike) -> bool:
+    return (
+        inspect.isasyncgenfunction(thing)
+        or inspect.isasyncgen(thing)
```
Comment (nfcampos) [inline langgraph/utils.py:154]: I don't think we need this one?

File: langgraph/utils.py (hunk 1 above)
Comment (hinthornw) [reply langgraph/utils.py:155]: tired brain. ya

File: langgraph/utils.py (hunk 2 above)
Comment (hinthornw) [reply langgraph/utils.py]: good catch

**Pull request merged**